components/VideoEmbedPreview.tsx

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import getYouTubeID from "get-youtube-id";
+
+function VideoEmbedPreview({ value }: { value: { url: string } }) {
+  const videoId = getYouTubeID(value.url);
+  const url = `https://www.youtube.com/embed/${videoId}`;
+
+  if (!videoId) {
+    return <p>Se necesita un enlace valido de Youtube.</p>;
+  }
+
+  return (
+    <iframe
+      width="520"
+      height="292.50"
+      src={url}
+      title="YouTube video player"
+      allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+      // allowFullScreen
```

Comment:
@TonyMckes @nykka 
Consideran necesario especificar `loading='lazy'`? [Loading lazy](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#attr-loading). Aunque para Firefox y Safari tanto en dektop como mobile no esta disponible.

Respecto al `accelerometer`, para que es necesario?

Estuve probando el `allowFullScreen` en un codepen en chrome y no me permite hacer el iframe full screen. Por lo que encontre en MDN este atributo fue redefinido como `fullscreen`. De esa manera si funciono, [allowfullscreen](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#attr-allowfullscreen).

Luego el `autoplay` lo consideran necesario? Personalmente me parece invasivo/molesto cuando ingresas a una pag. web y empieza el autoplay. Por cierto no logre que funcionara

El `width` no convendria setearlo igual que el container de mobile? 273px.

Reply:
Dude mucho en agregar un preview del video, mi idea iniciar era agregar el campo para el URL. Que les parece a ustedes? 

@LPRegen Sobre tus preguntas... todos esos atributos los provee Youtube, intentare responder igual..

- Me parece que `lazy loading` ya esta activado somehow.
- Supongo que `accelerometer` lo usa Youtube para dispositivos moviles (landscape, PiP).
- Comente `allowFullScreen` con la intencion de que sea solo un preview del video.
- Personalmente, me gusta `autoplay` en algunos casos, sobre todo cuando entro a un lugar justamente para ver un video jaja, pero si... no funciona aca.
- Sobre el `width` lo ideal seria hacerlo responsive, lo habia dejado asi para saber que pensaban de la idea de un video preview en Sanity.

Reply:
A mi me parece buena la idea de incluir el preview del video.
Pense que las habias agregado vos @TonyMckes .

- Por lo que entendi de la [documentacion](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy/accelerometer), tengo entendido que para el landscape utiliza el `gyroscope`.
- El `allowFullScreen`/`fullscreen` permite que el usuario haga fullscreen el video, me parece que esta bueno (especialmente en mobile).
- El `autoplay` como ustedes quieran :D. Aunque no me parece muy buena idea para alguien que utilice alguna tecnologia asistiva. Si se desea incluirlo, tambien se deberia reproducir el video sin sonido. Por cierto acabo de encontrar esto:  _This failure occurs when an audio or video element with an audio track contains the autoplay attribute and does not contain the muted attribute, and no controls or commands have been provided to pause or stop the media resource._ [Mas informacion](https://www.w3.org/TR/WCAG20-TECHS/F93.html)
- Me imagine que tenia que ser responsive, pero no se porque interprete que el `width` tiene que estar presente si o si en el elemento.

Reply:
Puede ser que la [Autoplay policy in Chrome](https://developer.chrome.com/blog/autoplay/) tambien tenga que ver.
Lo del `gyroscope` si tiene sentido, pero seguro que Youtube tenia sus motivos para agregarle el `accelerometer` xD

Sobre el `fullscreen` no se, la preview lo veo mas como situaciones de "ok, si agregue el URL correcto", en la pagina principal tendria mas sentido para mi, si

Reply:
Ah genial, interprete que se iba a utilizar para el FE.